Panic in reconcile with log level 7 and node on IPv6

/kind bug

**What steps did you take and what happened:**

- Set `-v=7` as an argument in CAPO's manifest and
- Add an IPv6 port in the OpenStackMachineTemplate
- CAPO crashes with `panic: key is not a string: 6`

**What did you expect to happen:**

CAPO should not crash.

**Anything else you would like to add:**

The issue comes from [this line](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/72e93bd4aabb5cd62c1aa7b9bd64ab98a37e614b/pkg/cloud/services/compute/instance_types.go#L160).
Our guess is that the panic refers to 'address.Version', rather than the verbosity level. In my case, address.Version == 6.

go-logr is used as the logger in this file, and go-logr does not support format strings like the code tries to use, see: https://github.com/go-logr/logr#but-i-really-want-to-use-a-format-string

Stack trace:
```
panic: key is not a string: 6

goroutine 441 [running]:
k8s.io/klog/v2/klogr.flatten({0xc000340a80, 0x2, 0xc0005fb250})
        /go/pkg/mod/k8s.io/klog/v2@v2.30.0/klogr/klogr.go:124 +0x4ff
k8s.io/klog/v2/klogr.klogger.Info({0x0, 0x1, {0xc0005c6040, 0x1b}, {0xc0005cc540, 0x10, 0x1c}, {0x183bd79, 0x9}}, 0x6, ...)
        /go/pkg/mod/k8s.io/klog/v2@v2.30.0/klogr/klogr.go:166 +0x493
github.com/go-logr/logr.Logger.Info({{0x1a8eac0, 0xc0004923c0}, 0xc0005c8470}, {0x18793de, 0x31}, {0xc0003409c0, 0x2, 0x2})
        /go/pkg/mod/github.com/go-logr/logr@v1.2.0/logr.go:249 +0xd0
sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/services/compute.(*InstanceStatus).NetworkStatus(0xc0003406e0)
        /workspace/pkg/cloud/services/compute/instance_types.go:160 +0x765
sigs.k8s.io/cluster-api-provider-openstack/controllers.(*OpenStackMachineReconciler).reconcileNormal(0x1a71138, {0x1a71138, 0xc0000dc0c0}, 0xc00008c380, 0xc00003d400, 0xc0005cc1c0, 0xc0005e0000, 0xc0005a0900, 0xc00003d400)
        /workspace/controllers/openstackmachine_controller.go:334 +0x51e
sigs.k8s.io/cluster-api-provider-openstack/controllers.(*OpenStackMachineReconciler).Reconcile(0xc0006c5c50, {0x1a71138, 0xc0000dc0c0}, {{{0xc000826ef0, 0x176adc0}, {0xc000826f60, 0x30}}})
        /workspace/controllers/openstackmachine_controller.go:161 +0xa5d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0000b6370, {0x1a71138, 0xc0000dc090}, {{{0xc000826ef0, 0x176adc0}, {0xc000826f60, 0x413c34}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0000b6370, {0x1a71090, 0xc000669240}, {0x16a8c00, 0xc0003408a0})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0000b6370, {0x1a71090, 0xc000669240})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:223 +0x357
```

**Environment:**

- Cluster API Provider OpenStack version: v0.6.2

--- a/pkg/cloud/services/compute/instance_types.go
+++ b/pkg/cloud/services/compute/instance_types.go
@@ -157,7 +157,7 @@ func (is *InstanceStatus) NetworkStatus() (*InstanceNetworkStatus, error) {
 
 			// Only consider IPv4
 			if address.Version != 4 {
-				is.logger.V(6).Info("Ignoring IPv%d address %s: only IPv4 is supported", address.Version, address.Address)
+				is.logger.V(6).Info("Ignoring IP address: only IPv4 is supported", "version", address.Version, "address", address.Address)
 				continue
 			}
 
@@ -168,7 +168,7 @@ func (is *InstanceStatus) NetworkStatus() (*InstanceNetworkStatus, error) {
 			case "fixed":
 				addressType = corev1.NodeInternalIP
 			default:
-				is.logger.V(6).Info("Ignoring address %s with unknown type '%s'", address.Address, address.Type)
+				is.logger.V(6).Info("Ignoring address with unknown type", "address", address.Address, "type", address.Type)
 				continue
 			}
 

--- a/pkg/cloud/services/compute/instance_types_test.go
+++ b/pkg/cloud/services/compute/instance_types_test.go
@@ -277,6 +277,37 @@ func TestInstanceNetworkStatus(t *testing.T) {
 			wantFloatingIP: "10.0.0.1",
 		},
 		{
+			name: "Ignore unknown address type",
+			addresses: map[string][]networkAddress{
+				"primary": {
+					{
+						Version: 4,
+						Addr:    "192.168.0.2",
+						Type:    "not-valid",
+						MacAddr: macAddr1,
+					}, {
+						Version: 4,
+						Addr:    "192.168.0.3",
+						Type:    "unknown",
+						MacAddr: macAddr2,
+					}, {
+						Version: 4,
+						Addr:    "10.0.0.1",
+						Type:    "floating",
+						MacAddr: macAddr3,
+					}, {
+						Version: 4,
+						Addr:    "192.168.0.1",
+						Type:    "fixed",
+						MacAddr: macAddr4,
+					},
+				},
+			},
+			networkName:    "primary",
+			wantIP:         "192.168.0.1",
+			wantFloatingIP: "10.0.0.1",
+		},
+		{
 			name: "Multiple networks",
 			addresses: map[string][]networkAddress{
 				"primary": {

